File: Graph/BFS of graph/BFS_of_graph.cpp
// Problem Name: BFS of graph
// Problem Link: https://www.geeksforgeeks.org/problems/bfs-traversal-of-graph/1

class Solution {
  public:
    // Function to return Breadth First Traversal of given graph.
    vector<int> bfs(vector<vector<int>>& adj,int V){
        vector<int> vis(V,0), ans;
        queue<int> q;
        q.push(0);
        vis[0] = 1;
        while(!q.empty()){
            int node = q.front();
            q.pop();
            ans.push_back(node);
            for(int it : adj[node]){
                if(!vis[it]){
                    vis[it] = 1;
                    q.push(it);
                }
            }
        }
        return ans;
    }
    
    vector<int> bfsOfGraph(vector<vector<int>> &adj) {
        // number of vertices/nodes in a graph
        int V = adj.size();
        return bfs(adj,V);
    }
};

File: Graph/BFS of graph/BFS_of_graph.java
// Problem Name: BFS of graph
// Problem Link: https://www.geeksforgeeks.org/problems/bfs-traversal-of-graph/1

class Solution {
    // Function to return Breadth First Traversal of given graph.
    public ArrayList<Integer> bfs(ArrayList<ArrayList<Integer>> adj, int V){
        ArrayList<Integer> ans = new ArrayList<Integer>();
        Queue<Integer> q = new LinkedList<>();
        boolean[] vis = new boolean[V];
        q.add(0);
        vis[0] = true;
        while (!q.isEmpty()) {
            Integer node = q.remove();
            ans.add(node);
            for (Integer it : adj.get(node)) {
                if (vis[it] == false) {
                    vis[it] = true;
                    q.add(it);
                }
            }
        }
        return ans;
    }
    public ArrayList<Integer> bfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj) {
        return bfs(adj,V);
    }
}